inc/post-templates.php

```diff
@@ -108,38 +112,80 @@ function is_post_template( $template = '' ) {
+ * and if the src is the same as the src as the post featured media image, or if
+ * the src are different but the attachment IDs are the same, then remove the first
+ * paragraph from the post's content to hide the duplicate image.
+ *
+ * This does catch img tags inside shortcodes.
+ *
+ * This does not remove leading images that are different from the post featured media
+ *
  * The changes to the content in this function should eventually be made
- * perminant in the database. (@see https://github.com/INN/Largo/issues/354)
- * 
- * @since 0.4
- * 
+ * permanent in the database. (@see https://github.com/INN/Largo/issues/354)
+ *
+ * If you would like to disable this function globally or on certain posts,
+ * use the filter `largo_remove_hero`.
+ *
+ * @since 0.4 - in Largo's single-column template
+ * @since 0.5.5 - in Largo's two-column template
+ *
  * @param String $content the post content passed in by WordPress filter
  * @return String filtered post content.
  */
-function largo_remove_hero($content) {
+function largo_remove_hero( $content ) {
 
 	global $post;
+	// Abort if there is no global $post
+	if ( !isset( $post ) ) {
+		return $content;
+	}
+
+	// 1: Only worry about this if:
+	// - it's a single template, and
+	// - there's a feature image, and
+	// - we haven't overridden the post display, and
+	// - we're not using a Largo layout
 
-	// 1: Only worry about this if it's a single template, there's a feature image,
-	// we haven't overridden the post display and we're not using the classic layout.
+	/**
+	 * Filter to disable largo_remove_hero based on the global $post at the time the function is run
+	 *
+	 * When building your own filter, you must set the fourth parameter of add_filter to 2:
+	 *
+	 *     function filter_largo_remove_hero( $run, $post ) {
+	 *         # determine whether or not to run largo_remove_hero based on $post
+	 *         return $run;
+	 *     }
+	 *     add_filter( 'largo_remove_hero', 'filter_largo_remove_hero', 10, 2 );
+	 *                                                                     ^
+	 *
+	 * @param Bool $run Whether `largo_remove_hero()` should be run
+	 * @param WP_Object $post The global $post at the time the function is run.
+	 * @since 0.5.5
+	 */
+	$do_run = apply_filters( 'largo_remove_hero', true, $post );
 
-	if( !is_single() ) 
+	if( !$do_run )
 		return $content;
 
-	if( !has_post_thumbnail() ) 
+	if( !has_post_thumbnail( $post->ID ) )
 		return $content;
 
-	$options = get_post_custom($post->ID);
+	$options = get_post_custom( $post->ID );
 
-	if( isset($options['featured-image-display'][0]) )
+	if( isset( $options['featured-image-display'][0] ) )
 		return $content;
 
-	if( of_get_option( 'single_template' ) == 'classic' )
+	if(
+		of_get_option( 'single_template' ) != 'normal' &&
+		of_get_option( 'single_template' ) != 'classic'
+	) {
 		return $content;
+	}
 
-	$p = explode("\n",$content);
+	$p = explode( '\n', $content );
```

Comment:
This is part of the reason the tests are now failing: This needs to be double quotes, to get the explode to work on newlines.